homeassistant/components/zha/api.py

```diff
@@ -326,8 +328,8 @@ async def websocket_get_device(
     hass: HomeAssistant, connection: ActiveConnection, msg: dict[str, Any]
 ) -> None:
     """Get ZHA devices."""
-    zha_gateway = hass.data[DATA_ZHA][DATA_ZHA_GATEWAY]
-    ieee = msg[ATTR_IEEE]
+    zha_gateway: ZHAGateway = hass.data[DATA_ZHA][DATA_ZHA_GATEWAY]
+    ieee: EUI64 = msg[ATTR_IEEE]
```

Comment:
Typing for `msg[ATTR_IEEE]` isn't perfect yet, since the validator `EUI64.convert` could also return `None` if `None` is passed. Thus technically all `ieee` variables should be `EUI64 | None`.

That is unpractical as almost always only `EUI64` is expected. For now I decided to use the optional variant only in combination with `msg.get(ATTR_IEEE)`.

Reply:
To followup on it, I checked all websocket endpoints and service calls. As expected, almost all values for `IEEE` are required and thus it's safe to assume `EUI64` for `ATTR_IEEE`. The exception here is the ´permit´ service / endpoint that is the only one for which it's optional.

PR to update the schema validators: #68353